janus_core/single_point.py

```diff
@@ -181,23 +183,33 @@ def run_single_point(
         properties : Optional[Union[str, list[str]]]
             Physical properties to calculate. If not specified, "energy",
             "forces", and "stress" will be returned.
+        write_kwargs : Optional[dict[str, Any]] = None,
+            Keyword arguments to pass to ase.io.write to save structure with
+            results of calculations. Must include "filename" keyword. Default is {}.
 
         Returns
         -------
         dict[str, Any]
             Dictionary of calculated results.
         """
-        results = {}
         if properties is None:
             properties = []
         if isinstance(properties, str):
             properties = [properties]
 
+        write_kwargs = write_kwargs if write_kwargs else {}
+        if write_kwargs and "filename" not in write_kwargs:
+            raise ValueError("'filename' must be included in write_kwargs")
+
+        results = {}
         if "energy" in properties or len(properties) == 0:
             results["energy"] = self._get_potential_energy()
         if "forces" in properties or len(properties) == 0:
             results["forces"] = self._get_forces()
         if "stress" in properties or len(properties) == 0:
             results["stress"] = self._get_stress()
 
+        if write_kwargs:
```

Comment:
i will have a special argument write_results... also there is an issue that will notice when you write for a non periodic system, stress is nan or inf

Reply:
Having it specified as `write_kwargs` means it matches writing out for geometry optimisation (final structure and/or trajectory), but yes I can add a `write_results` flag if we prefer that here.

In terms of `NaN`/`inf` values, to some extent isn't that something that should be handled by the calculator, rather than us? 

I hadn't realised until I added this, but even if we only run `get_potential_energy`, it sets all of the other values (forces, stress etc.) in `calc.results`, which makes sense in some ways, but it makes it more complex to write out, as we currently use the fact that it automatically writes out these `results`, so do we need to manually check these?